crates/re_data_source/src/data_loader/loader_directory.rs

```diff
@@ -0,0 +1,78 @@
+/// Loads entire directories, using the appropriate [`crate::DataLoader`]:s for each files within.
+//
+// TODO(cmc): There are a lot more things than can be done be done when it comes to the semantics
+// of a folder, e.g.: HIVE-like partitioning, similarly named files with different indices and/or
+// timestamps (e.g. a folder of video frames), etc.
+// We could support some of those at some point, or at least add examples to show users how.
+pub struct DirectoryLoader;
```

Comment:
You should make a note somewhere about wether or not the folder loader is recursive